docs/howtos/extracting_intermediates.rst

```diff
@@ -0,0 +1,272 @@
+Extracting intermediate values
+==============================
+
+This pattern will show you how to extract intermediate values from a module.
+Let's start with this simple CNN that uses :code:`nn.compact`.
+
+.. testsetup::
+
+  import flax.linen as nn
+  import jax
+  import jax.numpy as jnp
+  from flax.core import FrozenDict
+  from typing import Sequence
+
+  batch = jnp.ones((4, 32, 32, 3))
+
+  class SowCNN(nn.Module):
```

Comment:
I think this shouldn't be in the testsetup block as it's defined later again?

Reply:
The code in codediff blocks isn't currently part of the tests

Reply:
I see. Yeah @marcvanzee said he'll remove this once he brings in the change that tests the RHS of codediff blocks